admin/client/ColorSchemeDropdown.tsx

```diff
@@ -0,0 +1,129 @@
+import React = require("react")
+import { computed, action } from "mobx"
+import Select, { ValueType } from "react-select"
+import { ColorScheme, ColorSchemes } from "charts/ColorSchemes"
+import { observer } from "mobx-react"
+
+export interface ColorSchemeOption {
+    colorScheme?: ColorScheme
+    gradient?: string
+    label: string
+    value: string
+}
+
+export interface ColorSchemeDropdownProps {
+    additionalColorSchemes: ColorSchemeOption[]
+    defaultValue?: string
+    gradientColorCount: number
+    invertedColorScheme: boolean
+    onChange: (selected: ColorSchemeOption) => void
+}
+
+@observer
+export class ColorSchemeDropdown extends React.Component<
+    ColorSchemeDropdownProps
+> {
+    static defaultProps = {
+        additionalColorSchemes: [],
+        gradientColorCount: 6
+    }
+
+    @computed get availableColorSchemes() {
```

Comment:
@danielgavrilov Some debugging shows this (quite expensive) computed value is still re-calculated more times than is necessary.
I suspect this happens because MobX cannot infer that ColorSchemes didn't change in the meantime. Since you know React way better than I do, do you know of a way to prevent these unnecessary calculations? Can we make `ColorSchemes` observable somehow?

Reply:
Did you find out about [trace](https://mobx.js.org/best/trace.html) yet? It logs the cause for recomputing.

Reply:
Though you probably don't need to worry about efficiency in this part of the code, since it only runs on the admin side. 🙂